libs/template/config_test.go

```diff
@@ -398,3 +399,106 @@ func TestPromptIsSkipped(t *testing.T) {
 	assert.True(t, skip)
 	assert.Equal(t, "hello-world", c.values["xyz"])
 }
+
+func TestPromptIsSkippedAnyOf(t *testing.T) {
+	c := config{
+		ctx:    context.Background(),
+		values: make(map[string]any),
+		schema: &jsonschema.Schema{
+			Properties: map[string]*jsonschema.Schema{
+				"abc": {
+					Type: "string",
+				},
+				"def": {
+					Type: "integer",
+				},
+				"xyz": {
+					Type:    "string",
+					Default: "hello-world",
+					Extension: jsonschema.Extension{
+						SkipPromptIf: &jsonschema.Schema{
+							AnyOf: []*jsonschema.Schema{
+								{
+									Properties: map[string]*jsonschema.Schema{
+										"abc": {
+											Const: "foobar",
+										},
+										"def": {
+											Const: 123,
+										},
+									},
+								},
+								{
+									Properties: map[string]*jsonschema.Schema{
+										"abc": {
+											Const: "barfoo",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// No skip condition defined. Prompt should not be skipped.
+	skip, err := c.skipPrompt(jsonschema.Property{
+		Name:   "abc",
+		Schema: c.schema.Properties["abc"],
+	}, testRenderer())
+	assert.NoError(t, err)
+	assert.False(t, skip)
+
+	// Values do not match skip condition. Prompt should not be skipped.
+	c.values = map[string]any{
+		"abc": "foobar",
+		"def": 1234,
+	}
+	skip, err = c.skipPrompt(jsonschema.Property{
+		Name:   "xyz",
+		Schema: c.schema.Properties["xyz"],
+	}, testRenderer())
+	assert.NoError(t, err)
+	assert.False(t, skip)
+	assert.NotContains(t, c.values, "xyz")
+
+	// Missing values. Prompt should not be skipped.
+	c.values = map[string]any{
+		"abc": "foobar",
+	}
+	_, err = c.skipPrompt(jsonschema.Property{
+		Name:   "xyz",
+		Schema: c.schema.Properties["xyz"],
+	}, testRenderer())
+	assert.ErrorContains(t, err, "property def is used in skip_prompt_if but has no value assigned")
```

Comment:
I expected this test case to not return an error; `anyOf` contains an instance where only `abc` is set as being valid. Why is `def` also required?

Reply:
Reverted. Discussed offline that we don't want to have the "required" condition be explicit for `skip_prompt_if`